### PR TITLE
feat : added openPaymentPage that calls initiate and process together…

### PR DESCRIPTION
**Pull request by namitgoel**
… using HyperCheckoutLite openPaymentPage Function

### CODE REVIEW

File: android/src/main/kotlin/in/juspay/hyper_sdk_flutter/HyperSdkFlutterPlugin.kt [hunk 1]
```diff
@@ -145,6 +153,34 @@ class HyperSdkFlutterPlugin: FlutterPlugin, MethodCallHandler, ActivityAware, Pl
...
+        val callback = object : HyperPaymentsCallbackAdapter() {
+            override fun onEvent(data: JSONObject, p1: JuspayResponseHandler?) {
+                try {
+                    channel.invokeMethod(data.getString("event"), data.toString(), invokeMethodResult)
+                } catch (e: Exception) {
+                    Log.e(this.javaClass.canonicalName, "Failed to invoke method from native to dart", e)
+                }
+            }
+        }
+        HyperCheckoutLite.openPaymentPage(binding!!.activity as FragmentActivity, JSONObject(params), callback)
```
Comment (harshgarg18) [inline android/src/main/kotlin/in/juspay/hyper_sdk_flutter/HyperSdkFlutterPlugin.kt:181]: can we first add a check for whether activity is instance of FragmentActivity and throw an exception if it is not? You can reference initiate

File: android/build.gradle [hunk 2]
```diff
@@ -86,4 +86,5 @@ static def getClientId(rootProject) {
 hyperSdkPlugin {
     clientId = getClientId(rootProject)
     sdkVersion = getHyperSDKVersion(rootProject)
+    microSDKs = ['hypercheckoutlite']
```
[the remark this answers is not in the document]
Comment (namitgoel) [reply android/build.gradle]: @harshgarg18 done

File: android/src/main/kotlin/in/juspay/hyper_sdk_flutter/HyperSdkFlutterPlugin.kt [hunk 3]
```diff
@@ -75,14 +77,20 @@ class HyperSdkFlutterPlugin: FlutterPlugin, MethodCallHandler, ActivityAware, Pl
...
             else -> result.notImplemented()
         }
     }
 
     private fun onBackPress(result: Result) {
         try {
-            var backpress = hyperServices!!.onBackPressed()
-            result.success(backpress)
+            if (isHyperCheckOutLiteInteg) {
+                val backPress = HyperCheckoutLite.handleBackPress();
```
[the remark this answers is not in the document]
Comment (namitgoel) [reply android/src/main/kotlin/in/juspay/hyper_sdk_flutter/HyperSdkFlutterPlugin.kt]: @harshgarg18 got updated

File: android/src/main/kotlin/in/juspay/hyper_sdk_flutter/HyperSdkFlutterPlugin.kt (hunk 1 above)
Comment (namitgoel) [reply android/src/main/kotlin/in/juspay/hyper_sdk_flutter/HyperSdkFlutterPlugin.kt:185]: @harshgarg18 done

**Review by harshgarg18 — APPROVED** [review]

**Pull request merged**